remoteworkitem/scheduler.go

```diff
@@ -38,12 +40,26 @@ func (s *Scheduler) Stop() {
 // ScheduleAllQueries fetch and import of remote tracker items
 func (s *Scheduler) ScheduleAllQueries() {
 	cr.Stop()
+	ts := models.NewGormTransactionSupport(s.db)
+
 	trackerQueries := fetchTrackerQueries(s.db)
 	for _, tq := range trackerQueries {
 		cr.AddFunc(tq.Schedule, func() {
 			tr := LookupProvider(tq)
 			for i := range tr.Fetch() {
-				upload(s.db, tq.TrackerQueryID, i)
+				transaction.Do(ts, func() error {
+
+					// Save the remote items in a 'temporary' table.
+					err := upload(ts.TX(), tq.TrackerQueryID, i)
+					if err != nil {
+						return err
+					}
+
+					// Convert the remote item into a local work item and persist in the DB.
+					_, err = convert(ts, tq.TrackerQueryID, i, tq.TrackerType)
```

Comment:
Long term we need to know about the failure so it can be represented to user / ops etc. But ok for now.